pilot/pkg/proxy/envoy/v1/route.go

```diff
@@ -415,17 +405,17 @@ func buildHTTPRouteV2(store model.IstioConfigStore, config model.Config, service
 // TODO: This logic is temporary until we fully switch from v1alpha1 to v1alpha2.
 // In v1alpha2, cluster names will be built using the subset name instead of labels.
 // This will allow us to remove this function, which is very inefficient.
-func fetchSubsetLabels(store model.IstioConfigStore, name, subsetName, domain string) (labels model.Labels) {
+func fetchSubsetLabels(store model.IstioConfigStore, fqdn, subsetName, domain string) (labels model.Labels) {
```

Comment:
can we get rid of this function in subsequent PR?

Reply:
I'd love to. It will require a major refactor of how we build or cluster names / clusters though.